tests/network/docker-compose.yml

```diff
@@ -0,0 +1,140 @@
+    container_name: ocrd_network_mongo_db
+    networks:
+      - ocrd_network_test
+    ports:
+      - "6701:27017"
+    environment:
+      - MONGO_INITDB_ROOT_USERNAME=network_test
+      - MONGO_INITDB_ROOT_PASSWORD=network_test
+    healthcheck:
+      test: echo 'db.runCommand("ping").ok' | mongosh localhost:27017/test --quiet
+      interval: 1s
+      timeout: 3s
+      retries: 30
+
+  ocrd_network_rabbit_mq:
+    image: "rabbitmq:3.12-management"
+    hostname: rabbitmq-docker-host
+    container_name: ocrd_network_rabbit_mq
+    networks:
+      - ocrd_network_test
+    ports:
+      - "6672:5672"
+      - "16672:15672"
+      - "26672:25672"
+    environment:
+      - RABBITMQ_DEFAULT_USER=network_test
+      - RABBITMQ_DEFAULT_PASS=network_test
+      - RABBITMQ_FEATURE_FLAGS=quorum_queue,implicit_default_bindings,classic_mirrored_queue_version
+    healthcheck:
+      test: rabbitmq-diagnostics check_port_connectivity
+      interval: 1s
+      timeout: 3s
+      retries: 30
+
+  ocrd_network_processing_server:
+    image: "ocrd_core_test"
+    build:
+      context: ../../
+      dockerfile: Dockerfile
+      args:
+        BASE_IMAGE: 'ubuntu:22.04'
+      target: ocrd_core_test
+    hostname: processing-server-host
+    container_name: ocrd_network_processing_server
+    depends_on:
+      ocrd_network_mongo_db:
+        condition: service_healthy
+      ocrd_network_rabbit_mq:
+        condition: service_healthy
+    networks:
+      - ocrd_network_test
+    ports:
+      - "8000:8000"
+    environment:
+      DB_NAME: ocrd_network_test
+      DB_URL: mongodb://network_test:network_test@mongodb-docker-host.ocrd_network_test:27017
+      RABBITMQ_URL: amqp://network_test:network_test@rabbitmq-docker-host.ocrd_network_test:5672
+      OCRD_NETWORK_LOGS_ROOT_DIR: /ocrd-data/ocrd_network_logs
+      OCRD_NETWORK_SOCKETS_ROOT_DIR: /ocrd-data/ocrd_network_sockets
+    healthcheck:
+      test: curl -f http://processing-server-host.ocrd_network_test:8000/
+      interval: 1s
+      timeout: 3s
+      retries: 30
+    volumes:
+      - "/tmp/ocrd_network_logs:/ocrd-data/ocrd_network_logs"
+      - "/tmp/ocrd_network_sockets:/ocrd-data/ocrd_network_sockets"
+      - "./dummy-workflow.txt:/ocrd-data/assets/dummy-workflow.txt"
+      - "./ocrd_logging.conf:/etc/ocrd_logging.conf"
+    command: |
+      /bin/bash -c "echo -e \"
+        internal_callback_url: http://processing-server-host.ocrd_network_test:8000
+        process_queue:
+          address: rabbitmq-docker-host.ocrd_network_test
+          port: 5672
+          skip_deployment: true
+          credentials:
+            username: network_test
+            password: network_test
+        database:
+          address: mongodb-docker-host.ocrd_network_test
+          port: 27017
+          skip_deployment: true
+          credentials:
+            username: network_test
+            password: network_test
+        hosts: []\" > ./ps_config.yaml && \
```

Comment:
Could we make that somehow part of the `docker-compose.yml`. I see a potential for syntax errors and difficulty adapting when the config schema changes.

Reply:
Instead of doing that workaround, we could just volume map the `ps_config.yml`. However, the idea is that the credentials will be swapped with ENV variables. I have not found another optimal way to use environment variables inside the yaml file.

Reply:
I have decided to volume map the processing server configuration for now and keep adapting in two places. The processing server is no longer responsible for deploying RabbitMQ and MongoDB, according to the new spec, the config file is not needed in that case. The processing server can just get the MongoDB and RabbitMQ URLs as a parameter as processing workers do. However, the `ocrd_network` still needs to be refactored to reflect the latest spec changes.